<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>ADeclercq</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
.display.math{display: block; text-align: center; margin: 0.5rem auto;}
</style>


<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-html/tabby.min.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/light-border.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">


<link rel="stylesheet" href="styles.css">
</head>

<body>





<!-- Fonts -->
<link rel="preconnect" href="https://fonts.googleapis.com">
<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin="">
<link href="https://fonts.googleapis.com/css2?family=Cormorant+Garamond:ital,wght@0,300;0,400;0,500;0,600;0,700;1,300;1,400;1,500;1,600;1,700&amp;display=swap" rel="stylesheet">
<meta name="viewport" content="width=device-width, initial-scale=1">

<!-- favicon -->
<link rel="icon" type="image/png" href="./favicon-96x96.png" sizes="96x96">
<link rel="icon" type="image/svg+xml" href="./favicon.svg">
<link rel="shortcut icon" href="./favicon.ico">
<link rel="apple-touch-icon" sizes="180x180" href="./apple-touch-icon.png">
<meta name="apple-mobile-web-app-title" **content**="ADeclercq">
<link rel="manifest" href="./site.webmanifest">

<!-- Navigation Section -->
<nav class="navbar">
  <!-- Logo Section (Left-Aligned) -->
  <div class="logo-container">
    <img src="icons/AD_logo_full_colored_line.png" alt="Logo" class="logo">
  </div>

  <!-- Menu toggle checkbox -->
  <input type="checkbox" id="menu-toggle">
  <label for="menu-toggle" class="menu-icon">
      <div class="icon"></div>
  </label>

  <!-- Navigation Links Section -->
  <div class="nav-links">
    <a href="#" id="homeButton" class="nav-link">Home</a>
    <a href="#about" class="nav-link">About</a>
    <a href="#experience" class="nav-link">Experience</a>
    <a href="#publications" class="nav-link">Publications</a>
    <a href="#projects" class="nav-link">Projects</a>
    <a href="#contact" class="nav-link">Contact</a>
  </div>
</nav>

<!-- Home Section -->
<section id="home" class="home-section">
  <!-- Photographer Credit Box -->
  <div class="photo-credit">Photo by <a href="https://www.nathaliedolmans.be/" target="_blank">Nathalie Dolmans</a></div>
  <div class="home-intro">
    <h1 class="welcome-txt">Hi! My name is Arthur.</h1>
  </div>
</section>

<!-- Content Wrapper -->
<div id="main-content" class="main-content">

  <!-- About Section -->
  <section id="about" class="about-section max-widht">
    <div class="about-text">
      <h2 class="anchored">About Me</h2>
        <p>Hi! Before I explain that white thing I‚Äôm holding above, let me introduce myself. I'm Arthur Declercq, a PhD student at CompOmics, Ghent University. I graduated in Biomedical Sciences, where I discovered two things: a love for immunology and a knack for working with computers rather than lab equipment. This led me to bioinformatics, where I now develop AI algorithms to study proteins. My work focuses on identifying epitopes‚Äîtiny protein fragments shown on MHC molecules (yes, that‚Äôs the white thing!)‚Äîwhich are crucial for designing vaccines for cancer or fighting infections.</p>
        <p>When I'm not at my desk, you‚Äôll probably find me on the move. Running has become my passion, whether I‚Äôm racking up kilometers with the Ghent Running Crew or tackling trails. Beyond running, I love heading out with my tent for a backpacking adventure. And if you ever catch me sitting still, it‚Äôs likely with a good book in my hands.</p>
        <p>Reach out anytime if you‚Äôre into bioinformatics, outdoor adventures, or simply want to chat and exchange ideas!</p>
    </div>
  </section>

  <!-- Experience Section -->
  <section id="experience" class="experience-section">
    <div class="max-widht">
      <h2 class="anchored">Experience</h2>
      <div class="current-position">
        <h3 class="anchored">Current Position</h3>
        <p class="current-position-text"><strong>Predoctoral Researcher</strong> at <a href="https://vib.be/en/research-and-impact/research-centers/center-medical-biotechnology#/" target="_blank">VIB-UGent Center for Medical Biotechnology</a><br><span>2024 ‚Äì Present</span></p>
        <blockquote>"Working on data-driven bioinformatics solutions for mass spectrometry-based immunopeptidomics."</blockquote>
      </div>
    
  <!-- Education Section -->
      <div class="education-section">
        <h3 class="anchored">Education</h3>
        <ul class="education-list">
          <li>
            <strong>PhD in Bioinformatics</strong> ‚Äì Ghent University<br>
            <span>2021 ‚Äì ...</span>
          </li>
          <li>
            <strong>Deep learning specialisation</strong> ‚Äì Coursera<br>
            <span>2021 ‚Äì 2022</span>
          </li>
          <li>
            <strong>Master in Biomedical Sciences</strong> ‚Äì Ghent University<br>
            <span>2019 ‚Äì 2021</span>
          </li>
          <li>
            <strong>Ethical hacking from scratch in Python</strong> ‚Äì Udemy<br>
            <span>2020 ‚Äì 2021</span>
          </li>
          <li>
            <strong>Honours Programme in life sciences</strong> ‚Äì Ghent University<br>
            <span>2018 ‚Äì 2020</span>
          </li>
          <li>
            <strong>Bachelor in Biomedical Sciences</strong> ‚Äì Ghent University<br>
            <span>2016 ‚Äì 2019</span>
          </li>
        </ul>
      </div>

      <!-- Skills Section -->
      <div class="skills-section">
        <h3 class="anchored">Skills</h3>
        <div class="skills-list">
          <span class="skill-item">Python</span>
          <span class="skill-item">Machine Learning</span>
          <span class="skill-item">Immunopeptidomics</span>
          <span class="skill-item">Bioinformatics</span>
          <span class="skill-item">Proteomics</span>
          <span class="skill-item">Data Analysis</span>
        </div>
      </div>
    </div>
  </section>

  <!-- VOKA Section -->
  <section id="voka-section" class="voka-section">
    <div class="voka-background">
      <div class="photo-credit">
        Photo by 
        <a href="https://www.nathaliedolmans.be/" target="_blank">Nathalie Dolmans</a>
      </div>

      <div class="voka-quote">
        <h2 class="anchored">‚ÄúMeetinstrumenten in de biotechwereld worden elke dag beter, met onze software proberen we de ontwikkeling van mRNA-vaccins een duwtje in de rug te geven.‚Äù</h2>
        
        <!-- Link button with icon -->
        <a href="https://www.voka.be/nieuws/innovatiecafe-arthur-declercq-wint-2023-met-doctoraat-over-specifieke-epitopen-het-lichaam" class="voka-link">
          <span class="link-icon">üîó</span> <!-- Link icon -->
          Read Article
        </a>
      </div>
    </div>
  </section>


<!-- Publications Section -->
<section id="publications" class="publications-section">
  <div class="max-widht">
    <h2 class="publications-title anchored">Publications</h2>

    
    <!-- Tims¬≤Rescore -->
    <div class="publication-item">
      <h3 class="publication-title anchored">TIMS¬≤Rescore: A DDA-PASEF optimized data-driven rescoring pipeline based on MS¬≤Rescore</h3>
      <p class="publication-authors">
        <strong>Authors:</strong> Arthur Declercq ‚àô Robbe Devreese ‚àô Jonas Scheid ‚àô Caroline Jachmann ‚àô Tim Van Den Bossche ‚àô Annica Preikschat ‚àô David Gomez-Zepeda ‚àô Jeewan Babu Rijal ‚àô Aur√©lie Hirschler ‚àô Jonathan R Krieger ‚àô Tharan Srikumar ‚àô George Rosenberger ‚àô Dennis Trede ‚àô Christine Carapito ‚àô Stefan Tenzer ‚àô Juliane S Walz ‚àô Sven Degroeve ‚àô Robbin Bouwmeester ‚àô Lennart Martens ‚àô Ralf Gabriels 
      </p>
      <p class="publication-journal">
        <strong>Journal:</strong> Journal of Proteome Research<span>|</span> <strong>Year:</strong> February 2025 <span>|</span> 
        <span class="abstract-container" onclick="toggleAbstract(this)">
          Abstract
          <img src="icons/document.png" alt="abstract Icon" class="abstract-icon">
        </span>
      </p><div class="abstract">
        <p class="abstract-text">The high throughput analysis of proteins with mass spectrometry (MS) is highly valuable for understanding human biology, discovering disease biomarkers, identifying therapeutic targets, and exploring pathogen interactions. To achieve these goals, specialized proteomics subfields ‚Äì such as plasma proteomics, immunopeptidomics, and metaproteomics ‚Äì must tackle specific analytical challenges, such as an increased identification ambiguity compared to routine proteomics experiments. Technical advancements in MS instrumentation can counter these issues by acquiring more discerning information at higher sensitivity levels, as is exemplified by the incorporation of ion mobility and parallel accumulation - serial fragmentation (PASEF) technologies in timsTOF instruments. In addition, AI-based bioinformatics solutions can help overcome ambiguity issues by integrating more data into the identification workflow. Here, we introduce TIMS2Rescore, a data-driven rescoring workflow optimized for DDA-PASEF data from timsTOF instruments. This platform includes new timsTOF MS2PIP spectrum prediction models and IM2Deep, a new deep learning-based peptide ion mobility predictor. Furthermore, to fully streamline data throughput, TIMS2Rescore directly accepts Bruker raw mass spectrometry data, and search results from ProteoScape and many other search engines, including MS Amanda and PEAKS. We showcase TIMS2Rescore performance on plasma proteomics, immunopeptidomics (HLA class I and II), and metaproteomics data sets. TIMS2Rescore is open-source and freely available at <a href="https://github.com/compomics/tims2rescore" target="_blank" class="abstract-link">https://github.com/compomics/tims2rescore.</a></p>
        <a href="https://pubs.acs.org/doi/10.1021/acs.jproteome.4c00609" target="_blank" class="manuscript-link">
          View Manuscript
        </a>
      </div>
    </div>
    <div class="divider"></div>

    <!-- MHCquant 2 -->
    <div class="publication-item">
        <h3 class="publication-title anchored">MHCquant2 refines immunopeptidomics tumor antigen discovery</h3>
        <p class="publication-authors">
          <strong>Authors:</strong> Jonas Scheid ‚àô Steffen Lemke ‚àô Naomi Hoenisch-Gravel ‚àô Anna Dengler ‚àô Timo Sachsenberg ‚àô Arthur Declerq ‚àô Ralf Gabriels ‚àô Jens Bauer ‚àô Marcel Wacker ‚àô Leon Bichmann ‚àô Lennart Martens ‚àô Marissa L. Dubbelaar ‚àô Sven Nahnsen ‚àô Juliane S. Walz
        </p>
        <p class="publication-journal">
          <strong>Journal:</strong> Research Square<span>|</span> <strong>Year:</strong> December 2024 <span>|</span> 
          <span class="abstract-container" onclick="toggleAbstract(this)">
            Abstract
            <img src="icons/document.png" alt="abstract Icon" class="abstract-icon">
          </span>
        </p>
        <div class="abstract">
          <p class="abstract-text">The identification of human leukocyte antigen (HLA)-presented peptides as targets of anti-cancer T cell response is pivotal for the development of novel immunotherapies. Mass spectrometry (MS)-based immunopeptidomics enables the detection of these peptides, yet confident identifications and thus implementation in immunotherapy design are hampered by the high diversity and low abundance of naturally presented HLA peptides. Here, we introduce MHCquant2, a Nextflow-based open-source pipeline that leverages OpenMS tools and peptide property predictors (DeepLC, MS2PIP) for highly sensitive and scalable HLA peptide identification and quantification across various MS platforms. MHCquant2 increased peptide identifications up to 27% with a significant expansion of low-abundant peptides, outperforming state-of-the-art pipelines. Using MHCquant2 we build a comprehensive benign tissue repository comprising re-analyzed data from available benign immunopeptidomes and a novel benignMHCquant2 dataset, adding more than 160,000 novel naturally presented HLA peptides. First applications of this benign repository and the MHCquant2 pipeline enabled (i) the refinement of tumor-associated antigens, (ii) the detection of novel, high-frequent tumor-exclusive peptide antigens for multiple tumor entities, and (iii) the identification and quantification of mutation-derived low-abundant neoepitopes. MHCquant2 refines tumor antigen discovery in immunopeptidomics, paving the way for the implementation of off-the-shelf and personalized immunotherapy design.</p>
          <a href="https://www.researchsquare.com/article/rs-5560023/v1" target="_blank" class="manuscript-link">
            View Manuscript
          </a>
        </div>
      </div>
      <div class="divider"></div>
  
    <!-- Bacterial epitope discovery -->
    <div class="publication-item">
      <h3 class="publication-title anchored">Maximizing immunopeptidomics-based bacterial epitope discovery by multiple search engines and rescoring</h3>
      <p class="publication-authors">
          <strong>Authors:</strong> Patrick Willems ‚àô Fabien Thery ‚àô Laura Van Moortel ‚àô Margaux De Meyer ‚àô An Staes ‚àô Adillah Gul ‚àô Lyudmila Kovalchuke ‚àô Arthur Declercq ‚àô Robbe Devreese ‚àô Robbin Bouwmeester ‚àô Ralf Gabriels ‚àô  Lennart Martens ‚àô Francis Impens
      </p>
      <p class="publication-journal">
        <strong>Journal:</strong> Biorxiv<span>|</span> <strong>Year:</strong> November 2024 <span>|</span> 
        <span class="abstract-container" onclick="toggleAbstract(this)">
          Abstract
          <img src="icons/document.png" alt="abstract Icon" class="abstract-icon">
        </span>
      </p>
      <div class="abstract">
          <p class="abstract-text">Mass spectrometry-based discovery of bacterial immunopeptides presented by infected cells allows untargeted discovery bacterial antigens that can serve as vaccine candidates. Reliable identification of bacterial epitopes by such immunopeptidomics approaches is however challenged by their extreme low abundance. Here, we describe an optimized bioinformatical framework to enhance the confident identification of bacterial immunopeptides. Immunopeptidomics data of cell cultures infected with the foodborne model pathogen Listeria monocytogenes were searched by four different search engines, PEAKS, Comet, Sage and MSFragger, followed by data-driven rescoring with MS2Rescore. Compared to standard single search-engine results, this integrated workflow boosted the number of identified immunopeptides on average by 27% and led to the high confident detection of 18 additional bacterial peptides (+27%) matching 15 different Listeria proteins (+36%). Despite an overall large agreement between the search engines, a small number of conflicts (&lt; 1%) in spectra-to-peptide assignments revealed ambiguous identifications that served as a quality filter. Finally, we show compatibility of our workflow with sensitive timsTOF data acquisition and find that rescoring, now with inclusion of ion mobility features, identifies 76% more peptides compared to orbitrap-based acquisition. Together, our results demonstrate how integration of multiple search engine results along with data-driven rescoring maximizes the identification of immunopeptides, boosting the detection of high confident bacterial epitopes for vaccine development.</p>
          <a href="https://doi.org/10.1101/2024.11.22.624860" target="_blank" class="manuscript-link">
          View Manuscript
          </a>
      </div>
    </div>
    <div class="divider"></div>

    <!-- MS¬≤Rescore MCP -->
    <div class="publication-item">
      <h3 class="publication-title anchored">MS2Rescore: Data-Driven Rescoring Dramatically Boosts Immunopeptide Identification Rates</h3>
      <p class="publication-authors">
        <strong>Authors:</strong> Arthur Declercq ‚àô Robbin Bouwmeester ‚àô Aur√©lie Hirschler ‚àô Christine Carapito ‚àô Sven Degroeve ‚àô Lennart Martens ‚àô Ralf Gabriels
      </p>
      <p class="publication-journal">
        <strong>Journal:</strong> Molecular &amp; Cellular Proteomics<span>|</span> <strong>Year:</strong> August 2022 <span>|</span> 
        <span class="abstract-container" onclick="toggleAbstract(this)">
          Abstract
          <img src="icons/document.png" alt="abstract Icon" class="abstract-icon">
        </span>
      </p>
      <div class="abstract">
        <p class="abstract-text">Immunopeptidomics aims to identify major histocompatibility complex (MHC)-presented peptides on almost all cells that can be used in anti-cancer vaccine development. However, existing immunopeptidomics data analysis pipelines suffer from the nontryptic nature of immunopeptides, complicating their identification. Previously, peak intensity predictions by MS2PIP and retention time predictions by DeepLC have been shown to improve tryptic peptide identifications when rescoring peptide-spectrum matches with Percolator. However, as MS2PIP was tailored toward tryptic peptides, we have here retrained MS2PIP to include nontryptic peptides. Interestingly, the new models not only greatly improve predictions for immunopeptides but also yield further improvements for tryptic peptides. We show that the integration of new MS2PIP models, DeepLC, and Percolator in one software package, MS2Rescore, increases spectrum identification rate and unique identified peptides with 46% and 36% compared to standard Percolator rescoring at 1% FDR. Moreover, MS2Rescore also outperforms the current state-of-the-art in immunopeptide-specific identification approaches. Altogether, MS2Rescore thus allows substantially improved identification of novel epitopes from existing immunopeptidomics workflows.</p>
        <a href="https://www.mcponline.org/article/S1535-9476(22)00074-3/fulltext" target="_blank" class="manuscript-link">
          View Manuscript
        </a>
      </div>
    </div>
    <div class="divider"></div>

    <!-- MS¬≤Rescore MSAmanda -->
  <div class="publication-item">
  <h3 class="publication-title anchored">MS2Rescore 3.0 Is a Modular, Flexible, and User-Friendly Platform to Boost Peptide Identifications, as Showcased with MS Amanda 3.0</h3>
  <p class="publication-authors">
      <strong>Authors:</strong> Louise M. Buur ‚àô Arthur Declercq ‚àô Marina Strobl ‚àô Robbin Bouwmeester ‚àô Sven Degroeve ‚àô Lennart Martens ‚àô Viktoria Dorfer ‚àô Ralf Gabriels 
  </p>
  <p class="publication-journal">
      <strong>Journal:</strong> Journal of Proteome Research<span>|</span> <strong>Year:</strong> March 2024 <span>|</span> 
        <span class="abstract-container" onclick="toggleAbstract(this)">
          Abstract
          <img src="icons/document.png" alt="abstract Icon" class="abstract-icon">
        </span>
  </p><div class="abstract">
      <p class="abstract-text">Rescoring of peptide‚Äìspectrum matches (PSMs) has emerged as a standard procedure for the analysis of tandem mass spectrometry data. This emphasizes the need for software maintenance and continuous improvement for such algorithms. We introduce MS2Rescore 3.0, a versatile, modular, and user-friendly platform designed to increase peptide identifications. Researchers can install MS2Rescore across various platforms with minimal effort and benefit from a graphical user interface, a modular Python API, and extensive documentation. To showcase this new version, we connected MS2Rescore 3.0 with MS Amanda 3.0, a new release of the well-established search engine, addressing previous limitations on automatic rescoring. Among new features, MS Amanda now contains additional output columns that can be used for rescoring. The full potential of rescoring is best revealed when applied on challenging data sets. We therefore evaluated the performance of these two tools on publicly available single-cell data sets, where the number of PSMs was substantially increased, thereby demonstrating that MS2Rescore offers a powerful solution to boost peptide identifications. MS2Rescore‚Äôs modular design and user-friendly interface make data-driven rescoring easily accessible, even for inexperienced users. We therefore expect the MS2Rescore to be a valuable tool for the wider proteomics community. MS2Rescore is available at <a href="https://github.com/compomics/ms2rescore" target="_blank" class="abstract-link">https://github.com/compomics/ms2rescore.</a></p>
      <a href="https://pubs.acs.org/doi/10.1021/acs.jproteome.3c00785" target="_blank" class="manuscript-link">
      View Manuscript
      </a>
  </div>
  </div>
  <div class="divider"></div>

  <!-- MS¬≤PIP -->
  <div class="publication-item">
  <h3 class="publication-title anchored">Updated MS¬≤PIP web server supports cutting-edge proteomics applications</h3>
  <p class="publication-authors">
      <strong>Authors:</strong> Arthur Declercq ‚àô Robbin Bouwmeester ‚àô Cristina Chiva ‚àô Eduard Sabid√≥ ‚àô Aur√©lie Hirschler ‚àô Christine Carapito ‚àô Lennart Martens ‚àô Sven Degroeve ‚àô Ralf Gabriels 
  </p>
  <p class="publication-journal">
      <strong>Journal:</strong> Nucleic Acids Research<span>|</span> <strong>Year:</strong> July 2023 <span>|</span> 
        <span class="abstract-container" onclick="toggleAbstract(this)">
          Abstract
          <img src="icons/document.png" alt="abstract Icon" class="abstract-icon">
        </span>
  </p><div class="abstract">
      <p class="abstract-text">Interest in the use of machine learning for peptide fragmentation spectrum prediction has been strongly on the rise over the past years, especially for applications in challenging proteomics identification workflows such as immunopeptidomics and the full-proteome identification of data independent acquisition spectra. Since its inception, the MS¬≤PIP peptide spectrum predictor has been widely used for various downstream applications, mostly thanks to its accuracy, ease-of-use, and broad applicability. We here present a thoroughly updated version of the MS¬≤PIP web server, which includes new and more performant prediction models for both tryptic- and non-tryptic peptides, for immunopeptides, and for CID-fragmented TMT-labeled peptides. Additionally, we have also added new functionality to greatly facilitate the generation of proteome-wide predicted spectral libraries, requiring only a FASTA protein file as input. These libraries also include retention time predictions from DeepLC. Moreover, we now provide pre-built and ready-to-download spectral libraries for various model organisms in multiple DIA-compatible spectral library formats. Besides upgrading the back-end models, the user experience on the MS¬≤PIP web server is thus also greatly enhanced, extending its applicability to new domains, including immunopeptidomics and MS3-based TMT quantification experiments. MS¬≤PIP is freely available at <a href="https://iomics.ugent.be/ms2pip/." target="_blank" class="abstract-link">https://iomics.ugent.be/ms2pip.</a></p>
      <a href="https://academic.oup.com/nar/article/51/W1/W338/7151340?login=false" target="_blank" class="manuscript-link">
      View Manuscript
      </a>
  </div>
  </div>
  <div class="divider"></div>

  <!-- Thunder PASEF -->
  <div class="publication-item">
  <h3 class="publication-title anchored">Thunder-DDA-PASEF enables high-coverage immunopeptidomics and is boosted by MS2Rescore with MS2PIP timsTOF fragmentation prediction model</h3>
  <p class="publication-authors">
      <strong>Authors:</strong> David Gomez-Zepeda ‚àô Danielle Arnold-Schild ‚àô Julian Beyrle ‚àô Arthur Declercq ‚àô Ralf Gabriels ‚àô Elena Kumm ‚àô Annica Preikschat ‚àô Mateusz Krzysztof ≈ÅƒÖcki ‚àô Aur√©lie Hirschler ‚àô Jeewan Babu Rijal ‚àô Christine Carapito ‚àô Lennart Martens ‚àô Ute Distler ‚àô Hansj√∂rg Schild ‚àô Stefan Tenzer 
  </p>
  <p class="publication-journal">
      <strong>Journal:</strong> Nature Communications<span>|</span> <strong>Year:</strong> March 2024 <span>|</span> 
        <span class="abstract-container" onclick="toggleAbstract(this)">
          Abstract
          <img src="icons/document.png" alt="abstract Icon" class="abstract-icon">
        </span>
  </p><div class="abstract">
      <p class="abstract-text">Human leukocyte antigen (HLA) class I peptide ligands (HLAIps) are key targets for developing vaccines and immunotherapies against infectious pathogens or cancer cells. Identifying HLAIps is challenging due to their high diversity, low abundance, and patient individuality. Here, we develop a highly sensitive method for identifying HLAIps using liquid chromatography-ion mobility-tandem mass spectrometry (LC-IMS-MS/MS). In addition, we train a timsTOF-specific peak intensity MS2PIP model for tryptic and non-tryptic peptides and implement it in MS2Rescore (v3) together with the CCS predictor from ionmob. The optimized method, Thunder-DDA-PASEF, semi-selectively fragments singly and multiply charged HLAIps based on their IMS and m/z. Moreover, the method employs the high sensitivity mode and extended IMS resolution with fewer MS/MS frames (300‚Äâms TIMS ramp, 3 MS/MS frames), doubling the coverage of immunopeptidomics analyses, compared to the proteomics-tailored DDA-PASEF (100‚Äâms TIMS ramp, 10 MS/MS frames). Additionally, rescoring boosts the HLAIps identification by 41.7% to 33%, resulting in 5738 HLAIps from as little as one million JY cell equivalents, and 14,516 HLAIps from 20 million. This enables in-depth profiling of HLAIps from diverse human cell lines and human plasma. Finally, profiling JY and Raji cells transfected to express the SARS-CoV-2 spike protein results in 16 spike HLAIps, thirteen of which have been reported to elicit immune responses in human patients.</p>
      <a href="https://www.nature.com/articles/s41467-024-46380-y" target="_blank" class="manuscript-link">
      View Manuscript
      </a>
  </div>
  </div>
  <div class="divider"></div>

  <!-- Single cell bookchapter -->
  <div class="publication-item">
  <h3 class="publication-title anchored">Bioinformatics Pipeline for Processing Single-Cell Data</h3>
  <p class="publication-authors">
      <strong>Authors:</strong> Arthur Declercq ‚àô Nina Demeulemeester ‚àô Ralf Gabriels ‚àô Robbin Bouwmeester ‚àô Sven Degroeve ‚àô Lennart Martens
  </p>
  <p class="publication-journal">
      <strong>Book:</strong> Mass Spectrometry Based Single Cell Proteomics<span>|</span> <strong>Year:</strong> June 2024 <span>|</span> 
        <span class="abstract-container" onclick="toggleAbstract(this)">
          Abstract
          <img src="icons/document.png" alt="abstract Icon" class="abstract-icon">
        </span>
  </p><div class="abstract">
      <p class="abstract-text">Single-cell proteomics can offer valuable insights into dynamic cellular interactions, but identifying proteins at this level is challenging due to their low abundance. In this chapter, we present a state-of-the-art bioinformatics pipeline for single-cell proteomics that combines the search engine Sage (via SearchGUI), identification rescoring with MS2Rescore, quantification through FlashLFQ, and differential expression analysis using MSqRob2. MS2Rescore leverages LC-MS/MS behavior predictors, such as MS2PIP and DeepLC, to recalibrate scores with Percolator or mokapot. Combining these tools into a unified pipeline, this approach improves the detection of low-abundance peptides, resulting in increased identifications while maintaining stringent FDR thresholds.</p>
      <a href="https://link.springer.com/protocol/10.1007/978-1-0716-3934-4_15" target="_blank" class="manuscript-link">
      View Book Chapter
      </a>
  </div>
  </div>
  <div class="divider"></div>

  <!-- PSM Utils -->
  <div class="publication-item">
  <h3 class="publication-title anchored">Psm_utils: A high-level Python API for parsing and handling peptide-spectrum matches and proteomics search results</h3>
  <p class="publication-authors">
      <strong>Authors:</strong> Ralf Gabriels ‚àô Arthur Declercq ‚àô Robbin Bouwmeester ‚àô Sven Degroeve ‚àô Lennart Martens 
  </p>
  <p class="publication-journal">
      <strong>Journal:</strong> Journal of Proteome Research<span>|</span> <strong>Year:</strong> December 2022 <span>|</span> 
        <span class="abstract-container" onclick="toggleAbstract(this)">
          Abstract
          <img src="icons/document.png" alt="abstract Icon" class="abstract-icon">
        </span>
  </p><div class="abstract">
      <p class="abstract-text">A plethora of proteomics search engine output file formats are in circulation. This lack of standardized output files greatly complicates generic downstream processing of peptide-spectrum matches (PSMs) and PSM files. While standards exist to solve this problem, these are far from universally supported by search engines. Moreover, software libraries are available to read a selection of PSM file formats, but a package to parse PSM files into a unified data structure has been missing. Here, we present psm_utils, a Python package to read and write various PSM file formats and to handle peptidoforms, PSMs, and PSM lists in a unified and user-friendly Python-, command line-, and web-interface. psm_utils was developed with pragmatism and maintainability in mind, adhering to community standards and relying on existing packages where possible. The Python API and command line interface greatly facilitate handling various PSM file formats. Moreover, a user-friendly web application was built using psm_utils that allows anyone to interconvert PSM files and retrieve basic PSM statistics. psm_utils is freely available under the permissive Apache2 license at <a href="https://github.com/compomics/psm_utils" target="_blank" class="abstract-link">https://github.com/compomics/psm_utils.</a></p>
      <a href="https://pubs.acs.org/doi/abs/10.1021/acs.jproteome.2c00609" target="_blank" class="manuscript-link">
      View Manuscript
      </a>
  </div>
  </div>
  <div class="divider"></div>

  <!-- Ionmob -->
  <div class="publication-item">
  <h3 class="publication-title anchored">Ionmob: a Python package for prediction of peptide collisional cross-section values</h3>
  <p class="publication-authors">
      <strong>Authors:</strong> David Teschner ‚àô David Gomez-Zepeda ‚àô Arthur Declercq ‚àô Mateusz K ≈ÅƒÖcki ‚àô Seymen Avci ‚àô Konstantin Bob ‚àô Ute Distler ‚àô Thomas Michna ‚àô Lennart Martens ‚àô Stefan Tenzer ‚àô Andreas Hildebrandt 
  </p>
  <p class="publication-journal">
      <strong>Journal:</strong> Bioinformatics<span>|</span> <strong>Year:</strong> August 2023 <span>|</span> 
        <span class="abstract-container" onclick="toggleAbstract(this)">
          Abstract
          <img src="icons/document.png" alt="abstract Icon" class="abstract-icon">
        </span>
  </p><div class="abstract">
      <p class="abstract-text">Including ion mobility separation (IMS) into mass spectrometry proteomics experiments is useful to improve coverage and throughput. Many IMS devices enable linking experimentally derived mobility of an ion to its collisional cross-section (CCS), a highly reproducible physicochemical property dependent on the ion‚Äôs mass, charge and conformation in the gas phase. Thus, known peptide ion mobilities can be used to tailor acquisition methods or to refine database search results. The large space of potential peptide sequences, driven also by posttranslational modifications of amino acids, motivates an in silico predictor for peptide CCS. Recent studies explored the general performance of varying machine-learning techniques, however, the workflow engineering part was of secondary importance. For the sake of applicability, such a tool should be generic, data driven, and offer the possibility to be easily adapted to individual workflows for experimental design and data processing. We created ionmob, a Python-based framework for data preparation, training, and prediction of collisional cross-section values of peptides. It is easily customizable and includes a set of pretrained, ready-to-use models and preprocessing routines for training and inference. Using a set of ‚âà21‚Äâ000 unique phosphorylated peptides and ‚âà17‚Äâ000 MHC ligand sequences and charge state pairs, we expand upon the space of peptides that can be integrated into CCS prediction. Lastly, we investigate the applicability of in silico predicted CCS to increase confidence in identified peptides by applying methods of re-scoring and demonstrate that predicted CCS values complement existing predictors for that task. The Python package is available at github: <a href="https://github.com/theGreatHerrLebert/ionmob" target="_blank" class="abstract-link">https://github.com/theGreatHerrLebert/ionmob.</a></p>
      <a href="https://academic.oup.com/bioinformatics/article/39/9/btad486/7237255" target="_blank" class="manuscript-link">
      View Manuscript
      </a>
  </div>
  </div>
  <div class="divider"></div>

  <!-- PP plot -->
  <div class="publication-item">
  <h3 class="publication-title anchored">Quality control for the target decoy approach for peptide identification</h3>
  <p class="publication-authors">
      <strong>Authors:</strong> Elke Debrie ‚àô Milan Malfait ‚àô Ralf Gabriels ‚àô Arthur Declercq ‚àô Adriaan Sticker ‚àô Lennart Martens ‚àô Lieven Clement 
  </p>
  <p class="publication-journal">
      <strong>Journal:</strong> Journal of Proteome Research<span>|</span> <strong>Year:</strong> January 2023 <span>|</span> 
        <span class="abstract-container" onclick="toggleAbstract(this)">
          Abstract
          <img src="icons/document.png" alt="abstract Icon" class="abstract-icon">
        </span>
  </p><div class="abstract">
      <p class="abstract-text">Reliable peptide identification is key in mass spectrometry (MS) based proteomics. To this end, the target decoy approach (TDA) has become the cornerstone for extracting a set of reliable peptide-to-spectrum matches (PSMs) that will be used in downstream analysis. Indeed, TDA is now the default method to estimate the false discovery rate (FDR) for a given set of PSMs, and users typically view it as a universal solution for assessing the FDR in the peptide identification step. However, the TDA also relies on a minimal set of assumptions, which are typically never verified in practice. We argue that a violation of these assumptions can lead to poor FDR control, which can be detrimental to any downstream data analysis. We here therefore first clearly spell out these TDA assumptions, and introduce TargetDecoy, a Bioconductor package with all the necessary functionality to control the TDA quality and its underlying assumptions for a given set of PSMs.</p>
      <a href="https://pubs.acs.org/doi/abs/10.1021/acs.jproteome.2c00423" target="_blank" class="manuscript-link">
      View Manuscript
      </a>
  </div>
  </div>
  <div class="divider"></div>


  <!-- cross-links -->
  <div class="publication-item">
  <h3 class="publication-title anchored">Intensity and retention time prediction improves the rescoring of protein-nucleic acid cross-links</h3>
  <p class="publication-authors">
      <strong>Authors:</strong> Arslan Siraj ‚àô Robbin Bouwmeester ‚àô Arthur Declercq ‚àô Luisa Welp ‚àô Aleksandar Chernev ‚àô Alexander Wulf ‚àô Henning Urlaub ‚àô Lennart Martens ‚àô Sven Degroeve ‚àô Oliver Kohlbacher ‚àô Timo Sachsenberg 
  </p>
  <p class="publication-journal">
      <strong>Journal:</strong> Proteomics<span>|</span> <strong>Year:</strong> April 2024 <span>|</span> 
        <span class="abstract-container" onclick="toggleAbstract(this)">
          Abstract
          <img src="icons/document.png" alt="abstract Icon" class="abstract-icon">
        </span>
  </p><div class="abstract">
      <p class="abstract-text">In protein-RNA cross-linking mass spectrometry, UV or chemical cross-linking introduces stable bonds between amino acids and nucleic acids in protein-RNA complexes that are then analyzed and detected in mass spectra. This analytical tool delivers valuable information about RNA-protein interactions and RNA docking sites in proteins, both in vitro and in vivo. The identification of cross-linked peptides with oligonucleotides of different length leads to a combinatorial increase in search space. We demonstrate that the peptide retention time prediction tasks can be transferred to the task of cross-linked peptide retention time prediction using a simple amino acid composition encoding, yielding improved identification rates when the prediction error is included in rescoring. For the more challenging task of including fragment intensity prediction of cross-linked peptides in the rescoring, we obtain, on average, a similar improvement. Further improvement in the encoding and fine-tuning of retention time and intensity prediction models might lead to further gains, and merit further research.</p>
      <a href="https://analyticalsciencejournals.onlinelibrary.wiley.com/doi/full/10.1002/pmic.202300144" target="_blank" class="manuscript-link">
      View Manuscript
      </a>
  </div>
  </div>
  <div class="divider"></div>
  </div>
  </section>

  <script>
  function toggleAbstract(element) {
    // Select the clicked abstract
    const abstract = element.closest('.publication-item').querySelector('.abstract');

    // Close all other abstracts
    document.querySelectorAll('.abstract').forEach(item => {
      if (item !== abstract) item.style.display = 'none';
    });

    // Toggle the clicked abstract
    abstract.style.display = abstract.style.display === 'block' ? 'none' : 'block';
  }
  </script>


  <!-- Projects Section -->
  <section id="projects" class="projects-section max-widht">
    <h2 class="anchored">Projects</h2>
    <!-- Card Container -->
    <div class="projects-container">
      <!-- MS¬≤Rescore -->
      <div class="project-card">
        <h3 class="project-title anchored">MS¬≤Rescore</h3>
        <p class="project-description">Modular and user-friendly platform for AI-assisted rescoring of peptide identifications.</p>
        <a href="https://github.com/compomics/ms2rescore" target="_blank" class="project-link">github.com/compomics/ms2rescore</a>
      </div>

      <!-- MS¬≤PIP -->
      <div class="project-card">
        <h3 class="project-title anchored">MS¬≤PIP</h3>
        <p class="project-description">MS2 Peak Intensity Prediction - Fast and accurate peptide fragmentation spectrum prediction.</p>
        <a href="https://github.com/compomics/ms2pip" target="_blank" class="project-link">github.com/compomics/ms2pip</a>
      </div>

      <!-- Mumble -->
      <div class="project-card">
        <h3 class="project-title anchored">Mumble</h3>
        <p class="project-description">Mumble is a Python-based tool designed to find candidate unimod modifications for mass shifts.</p>
        <a href="https://github.com/compomics/mumble" target="_blank" class="project-link">github.com/compomics/mumble</a>
      </div>

      <!-- MHCMolPred -->
      <div class="project-card">
        <h3 class="project-title anchored">MHC-3PO</h3>
        <p class="project-description">MHC binding affinity predictions based using large language models and graph neural nets.</p>
        <span class="project-coming-soon">Coming soon</span>
      </div>

      <!-- Baxerna -->
      <div class="project-card">
        <h3 class="project-title anchored">Baxerna</h3>
        <p class="project-description">Contributor to Baxerna, a next-generation bacterial vaccine development pipeline to generate novel pathogen-specific mRNA vaccines.</p>
        <a href="https://www.baxerna.eu/" target="_blank" class="project-link">baxerna.eu</a>
      </div>

      <!-- Website -->
      <div class="project-card">
        <h3 class="project-title anchored">Website</h3>
        <p class="project-description">Creating my own website using the Quarto framework.</p>
        <a href="https://github.com/ArthurDeclercq/arthurdeclercq.github.io" target="_blank" class="project-link">github.com/arthurdeclercq.github.io</a>
      </div>
    </div>
</section>


<!-- Contact Section -->
<section id="contact" class="contact-section">
  <!-- Background and Overlay -->
  <div class="contact-overlay"></div>
  <div class="max-widht">
    <!-- Contact Content -->
    <div class="contact-content">
      <h2 class="contact-title anchored">Get in touch</h2>

      <!-- Full Name and Logo -->
      <div class="contact-name">
        <img src="icons/AD_logo_small.png" alt="Logo" class="contact-logo">
        <p class="contact-full-name">Arthur Declercq</p>
      </div>

      <!-- Contact Information -->
      <div class="contact-info">
        <!-- Email -->
        <div class="contact-item">
          <img src="icons/send.png" alt="Email Icon" class="contact-icon">
          <a href="mailto:Arthur.Declercq@UGent.be" class="contact-link">Arthur.Declercq@UGent.be</a>
        </div>

        <!-- LinkedIn -->
        <div class="contact-item">
          <img src="icons/linkedin.png" alt="LinkedIn Icon" class="contact-icon">
          <a href="https://www.linkedin.com/in/arthur-declercq-b37b9b1a4/" target="_blank" class="contact-link">linkedin.com/in/arthur-declercq</a>
        </div>

        <!-- GitHub -->
        <div class="contact-item">
          <img src="icons/github.png" alt="GitHub Icon" class="contact-icon">
          <a href="https://github.com/ArthurDeclercq" target="_blank" class="contact-link">github.com/ArthurDeclercq</a>
        </div>

        <!-- Google Scholar -->
        <div class="contact-item">
          <img src="icons/google.png" alt="Google Scholar Icon" class="contact-icon">
          <a href="https://scholar.google.com/citations?user=your-google-scholar-id" target="_blank" class="contact-link">Google Scholar/ArthurDeclercq</a>
        </div>
j
        <!-- Twitter
        <div class="contact-item">
          <img src="icons/x.png" alt="x Icon" class="contact-icon">
          <a href="https://x.com/DeclercqArthur" target="_blank" class="contact-link">x.com/DeclercqArthur</a>
        </div> -->

        <!-- Orcid -->
        <div class="contact-item">
          <img src="icons/orcid_white.png" alt="Orcid Icon" class="contact-icon">
          <a href="https://orcid.org/0000-0002-9376-1399" target="_blank" class="contact-link">orcid.org/0000-0002-9376-1399</a>
        </div>
      </div>
    </div>
  </div>
</section>

</div>

<script>
document.addEventListener('DOMContentLoaded', function() {
  // Get all the navigation links in the navbar
  var navLinks = document.querySelectorAll('.nav-links a');
  var menuToggle = document.getElementById('menu-toggle');

  // Add click event listeners to navbar links only
  navLinks.forEach(function(link) {
    link.addEventListener('click', function(event) {
      // Prevent default anchor behavior
      event.preventDefault();

      // Check if the link is the "Home" button
      if (this.getAttribute('id') === 'homeButton') {
        // Scroll to the top smoothly
        window.scrollTo({ top: 0, behavior: 'smooth' });
      } else {
        // For other links, scroll to the target section
        var targetId = this.getAttribute('href').substring(1);
        var targetElement = document.getElementById(targetId);

        // Calculate the scroll position, accounting for the navbar height
        var navbarHeight = document.querySelector('.navbar').offsetHeight;
        var targetPosition = targetElement.getBoundingClientRect().top + window.pageYOffset - navbarHeight;

        // Smoothly scroll to the target position
        window.scrollTo({ top: targetPosition, behavior: 'smooth' });
      }

      // Uncheck the menu toggle checkbox to close the menu
      menuToggle.checked = false;
    });
  });
});
</script>

<script>
  document.addEventListener("DOMContentLoaded", function () {
    let today = new Date();
    let newThreshold = 200; // Days to consider an item as new
    let publications = document.querySelectorAll(".publication-item");

    publications.forEach((pub) => {
      let yearText = pub.querySelector(".publication-journal").innerText.match(/\b\d{4}\b/);
      let monthText = pub.querySelector(".publication-journal").innerText.match(/(January|February|March|April|May|June|July|August|September|October|November|December)/);

      if (yearText && monthText) {
        let pubDate = new Date(`${monthText[0]} 1, ${yearText[0]}`);
        let diffDays = (today - pubDate) / (1000 * 60 * 60 * 24);

        if (diffDays <= newThreshold) {
          let newTag = document.createElement("span");
          newTag.classList.add("new-tag");
          newTag.textContent = "New";

          let title = pub.querySelector(".publication-title");
          if (title) {
            title.appendChild(newTag);
          }
        }
      }
    });
  });
</script>



<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
  tabsets.forEach(function(tabset) {
    const tabby = new Tabby('#' + tabset.id);
  });
  const icon = "Óßã";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/arthurdeclercq\.bio");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'light-border',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>




</body></html>